Add virtual-pet extension

timoransky opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [ ] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [ ] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [ ] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ ] I checked that files in the `assets` folder are used by the extension itself
- [ ] I checked that assets used by the `README` are placed outside of the `metadata` folder

timoransky commented on extensions/virtual-pet/src/hooks/usePetState.ts:
```diff
@@ -0,0 +1,58 @@
+      return;
+    }
+
+    const newState = action(petState);
+    setPetState(newState);
```
This is intentional

timoransky commented:
I implemented some feedback from the AI bot which makes sense in my opinion and also pushed some balancing fixes from my testing. Let me know if there is anything else I should take a look into

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Status doesn’t match in Menu Bar compared to the view command

![CleanShot 2025-03-17 at 14 06 06@2x](https://github.com/user-attachments/assets/7ffe8ae2-516a-49c2-86d7-97d4fc5201a1)

- When your pet is all dead, should we consider showing the adoption screen again?

![virtual-pet 2025-03-17 at 14 12 15](https://github.com/user-attachments/assets/395367ce-d126-4bea-b7e8-a644a8db2156)


I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

timoransky commented:
Hey @pernielsentikaer ,

Thanks for looking at this PR. Let me explain my thinking process behind both of your points, maybe we can brainstorm a bit about these as well...

1. The status tags don't match up because those which are visible in the menu bar are only the critical ones (if the values are 40 or bellow) to highlight that the pet requires more attention. On the other hand, in the detail view I display more detailed statuses, just for the information and fun purposes, to keep the interaction between the user and the extension more dynamic.
2. Regarding the pet dying... This was something that I already implemented previously, but during my testing I realized it was not really working properly in a sense of a long term feature. Currently I don't account for the Mac sleep time, so when you wake your Mac after the night of sleep, the extension will check last updated time which is yesterday, it will do its calculations and you will found your pet dead. I didn't really like this, because it would most likely mean that your pet can die really quickly, so I implemented the "Heal" action instead. This can be used only once per day and it recovers your pet a little, so you can continue taking care of it. But I want to get back to this idea later, once I gather some feedback from more users.

I hope this answers your questions, if you have any other feedback, or if you want to discuss something in more detail please do let me know 🙂

pernielsentikaer commented:
That makes a lot of sense, thanks for the thorough explanations 😊

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
